test/TestCategory.cpp

```diff
@@ -0,0 +1,34 @@
+#include "../src/Category.h"
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+
+TEST(TestCategory, TestCategory) {
+  Category category1("cat0", 0);
+  std::ostringstream os;
+  os << category1;
+  EXPECT_EQ(os.str(), "Category(cat0:0)");
+}
+
+TEST(TestCategory, TestCategory2) {
+  Category category1("cat12", 12);
+  std::ostringstream os;
+  os << category1;
+  EXPECT_EQ(os.str(), "Category(cat12:12)");
+}
+
+
+TEST(TestCategory, TestConstructorByCopy) {
+  Category category1("cat1", 13);
+
+  Category catCopied = Category(category1);
+
+  std::ostringstream os;
+  os << catCopied;
+  EXPECT_EQ(os.str(), "Category(cat1:13)");
+}
+
+TEST(TestCategory, TestAllInstancesDestroyed) {
+  EXPECT_EQ(AtomicMCDAObject::get_nb_instances(), 0);
```

Comment:
Not 100% sure how it work but I added `int AtomicMCDAObject::nb_instances_{0};` at the begining of my file, I think we need that because if there are other destructors failing (and thus nb_instances not at 0) it will also fail on that test and make things a bit confusing

Reply:
added AtomicMCDAObject nb instance in the TestCategory